core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -200,7 +210,101 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
   }
 
   @Override
-  public void close() throws IOException {
+  public void createNamespace(Namespace namespace, ImmutableMap<String, String> meta) {
+    Preconditions.checkArgument(
+        !namespace.isEmpty(),
+        "Cannot create namespace with invalid name: %s", namespace);
+    if (!meta.isEmpty()) {
+      throw new UnsupportedOperationException("Cannot create namespace " + namespace + " : metadata is not supported");
+    }
+
+    Path nsPath = new Path(warehouseLocation, SLASH.join(namespace.levels()));
+    FileSystem fs = Util.getFs(nsPath, conf);
+
+    try {
+      if (isNamespace(fs, nsPath)) {
+        throw new AlreadyExistsException("Namespace '%s' already exists!", namespace);
+      }
+
+      fs.mkdirs(nsPath);
+
+    } catch (IOException e) {
+      throw new RuntimeIOException(e, "Create namespace failed: %s", namespace);
+    }
   }
 
+  @Override
+  public List<Namespace> listNamespaces(Namespace namespace) {
+    Path nsPath = new Path(SLASH.join(warehouseLocation, SLASH.join(namespace.levels())));
```

Comment:
Please fix path construction here and other places that need it as well.

Reply:
OK